exercises/transpose/transpose_test_with_heredoc.rb

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env ruby
+gem 'minitest', '>= 5.0.0'
+require 'minitest/autorun'
+require_relative 'transpose'
+
+# Test data version:
+# 0a51cfc
+class TransposeTest < Minitest::Test
+
+  def test_empty_string
+    # skip
+    input = <<-INPUT
+
+INPUT
+    actual = Transpose.transpose(input)
+    expected = <<-EXPECTED
+
+EXPECTED
+    assert_equal expected.strip, actual
+  end
+
+  def test_two_characters_in_a_row
+    skip
+    input = <<-INPUT
+A1
+INPUT
+    actual = Transpose.transpose(input)
+    expected = <<-EXPECTED
+A
+1
+EXPECTED
+    assert_equal expected.strip, actual
+  end
+
+  def test_two_characters_in_a_column
+    skip
+    input = <<-INPUT
+A
+1
+INPUT
+    actual = Transpose.transpose(input)
+    expected = <<-EXPECTED
+A1
+EXPECTED
+    assert_equal expected.strip, actual
+  end
+
+  def test_simple
+    skip
+    input = <<-INPUT
+ABC
+123
+INPUT
```

Comment:
With a tilde `~` in front of the Heredoc delimiter, you can present these heredocs with the proper indentation, and they will end up looking good here and clean in the generated files.  Including indentation in side the text.  This does require Ruby 2.3 though, but it is not a hard thing to write.  Facets gem has this functionality for a fairly long time (I wrote one in 2008, unsure if Facets got my version).